File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
from collections import defaultdict

def solution(id_list, report, k):
    answer = [0]*len(id_list)
    id_list = dict(zip(id_list, [i for i in range(len(id_list))]))
    dic = defaultdict(set)
    for r in report:
        x, y = r.split()
        dic[y].add(x)
    for key in dic:
        if len(dic[key])>=k:
            for val in dic[key]:
                answer[id_list[val]] += 1
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        
        List<String[]> distinct_report = Arrays.stream(report).distinct().map(x -> x.split(" ")).collect(Collectors.toList());
        
        HashMap<String, Integer> reported_users = new HashMap<>();
        
        for(String[] users: distinct_report) {
            reported_users.put(users[1], reported_users.getOrDefault(users[1], 0) + 1);
        }
        
        HashMap<String, Integer> reporting_users = new HashMap<>();
        
        for(String user: id_list) {
            reporting_users.put(user, 0);
        }
        
        for(String[] users: distinct_report) {
            if(reported_users.get(users[1]) >= k) {
                reporting_users.put(users[0], reporting_users.get(users[0]) + 1);
            }
        }
        
        int len = id_list.length;
        int[] answer = new int[len];
        for(int i=0 ; i<len; i++) {
            answer[i] = reporting_users.get(id_list[i]).intValue();
        }
        
        return answer;
    }
}